### PR TITLE
Possible Shibboleth fix.

### PR DESCRIPTION
**Pull request by bwatson78**

### CODE REVIEW

File: app/controllers/omniauth_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class OmniauthController < Devise::SessionsController
   def new
     # Rails.logger.debug "SessionsController#new: request.referer = #{request.referer}"
     if Rails.env.production?
-      redirect_to user_shibboleth_omniauth_authorize_path
+      redirect_post(user_shibboleth_omniauth_authorize_path, options: { authenticity_token: :auto })
```
[the remark this answers is not in the document]
Comment (bwatson78) [reply app/controllers/omniauth_controller.rb:7]: I'll just steal the code from the gem-no worries.

[the remark this answers is not in the document]
Comment (bwatson78) [reply app/controllers/omniauth_controller.rb:7]: @abelemlih I've been digging through `repost`'s code and am realizing it isn't at all simple to copy its code into our application. We are receiving the `Not found. Authentication passthru.` error when trying to log into Curate via Shibboleth. The reason this is happening is that the `redirect_to` command is inherently a GET call, not a POST call that devise v2.* now requires for all authorization controller methods. 

The issue with that is that all redirects in rails are GETs. This gem properly formats the call to this link with all of the appropriate params and attributes passed along with a dynamically created post call. I hope this answers the questions you have, but feel free to sift through`repost`'s code to see exactly what it does.

[the remark this answers is not in the document]
Comment (abelemlih) [reply app/controllers/omniauth_controller.rb:7]: @bwatson78 Can you please look into the issues raised by Code Climate? I will review the PR again once it passes all checks.

**Comment by bwatson78** [comment]
@abelemlih Would you consider approving the current code? The diff-coverage analysis won't budge unless I make a test directly associated to the code I changed, which doesn't make much sense, in my opinion.

**Comment by abelemlih** [comment]
@bwatson78 have we tried locking Omniauth to `~> 1.0.0`? the `omniauth-shibboleth` gem which we currently use to setup the shibboleth strategy is locked to `'>= 1.0.0'`, so having our application currently upgraded to `omniauth (2.1.0)` may be the reason behind the issues we are seeing.

In any case, we may want to consider doing the `omniauth` upgrade in a separate PR, and ensuring the app works as expected with Omniauth 1.X.

**Comment by bwatson78** [comment]
The reason why we bumped up to Omniauth v2.* was to eliminate security vulnerabilities---wouldn't have touched it if not to correct those vulnerabilities.

**Comment by abelemlih** [comment]
@bwatson78 I assume you are possibly referring to vulnerabilities in `omniauth v1.9.1`, which a patch was released for in `v1.9.2`, https://github.com/omniauth/omniauth/releases/tag/v1.9.2. Would `v1.9.2` be a good enough upgrade to resolve the immediate security vulnerabilities you are referring to?

**Comment by bwatson78** [comment]
@abelemlih I went to Omniauth v2.* at the recommendation of the Github security alert. Didn't try other versions because the alert said that the versions to migrate to--the versions that would eliminate the alert--would be v2 or higher.

**Comment by abelemlih** [comment]
I understand that, I just think we have to be careful about the following:

- Adding a third party dependency, repost , for the sole reason of redirecting to the omniauth page with a post request. Having a third party dependency in our login workflow is always a risk.
- Upgrading from Omniauth 1 to 2 should ideally be a project where we look carefully into our login workflow, and determine what code is still relevant, and what is deprecated. The library we currently use for the omniauth strategy is build specifically for Omniauth 1, so using it with Omniauth 2 seems risky.

Feel free to include @alexBLR  or @rotated8  in the PR and get their opinions too. I am being extra careful here, and if they deem it unnecessary, I am fine with that.

**Comment by bwatson78** [comment]
@rotated8 Your thoughts?

**Comment by abelemlih** [comment]
@bwatson78 @rotated8 a few things to note:
- Omniauth v2 is the version that will be receiving patches moving forward, so we will have to upgrade soon anyways.
- Looking at our code, we may want to consider rewriting our omniauth related code to avoid using libraries like `omniauth-shibboleth` that have not followed through with an upgrade to Omniauth v2.

**Comment by rotated8** [comment]
Let's go back to `omniauth` 1.9.x for now. That fixes `main` and allows work to continue. I don't have security concerns about `repost`, but I don't want us to get into the position of needing to patch up our authentication when the gems we rely on aren't updating.

Since `omniauth-shibboleth` says it is a strategy for `omniauth` 1.x, we should move to `omniauth-saml` as we upgrade to `omniauth` 2.x, but we can do that more more slowly.

**Pull request closed without merging**